kivy/core/camera/camera_avfoundation.pyx

```diff
@@ -105,7 +119,10 @@ class CameraAVFoundation(CameraBase):
 
     def set_framerate(self, framerate):
         cdef _AVStorage storage = <_AVStorage>self._storage
-        avf_camera_attempt_framerate_selection(storage.camera, framerate)
+        if avf_camera_attempt_framerate_selection(storage.camera, framerate):
+            if self._update_ev is not None:
+                self._update_ev.cancel()
+            self._update_ev = Clock.schedule_interval(self._update, 1 / float(framerate))
```

Comment:
You're better off scheduling it at e.g. 4 times the framerate. Because if a frame is not available, it'll just return early. But you don't want to be waiting for a frame if one is available. I'd also change it for the 1 / 30 case.

Reply:
It makes sense, but We need to make a couple of (simple?) new edits to the provider, in order to stop the update process immediately if it's not needed.

Without any change, multiplying the updates to x4 the frame rate will request to handle x4 times all the update process that is huge.

As an example, now, after these changes I made, on an iPhone 12 Pro, @ 30 FPS, I'm having a 30/35 % of constant CPU usage (When using the default Preset in High Res). I don't imagine doing it x4 times.

P.S: Now, after these changes all the CPU usage seems to be on `glTexImage2D` (I noticed that via XCode Instruments)

Reply:
Hmm, but I don't expect the CPU would be working 4 times as harder. Rather kivy will just execute the callback 4 times as much, but on average, 3 out of those 4 times it'll return immediately with no new frame? 

I expect the high CPU is coming from reading the actual frames, and although you'd lose less frames, I expect the CPU increase will be minimal, proportional to the increase in frames you're not missing. But it shouldn't increase 4x. And if the CPU is too much, then decreasing the frame rate should be the solution, not missing frames. I.e. ideally no frame is missed.

But I get it, this can be done in a future PR, not need to troubleshoot it all in this PR.

Reply:
> Hmm, but I don't expect the CPU would be working 4 times as harder. Rather kivy will just execute the callback 4 times as much, but on average, 3 out of those 4 times it'll return immediately with no new frame?

That's the point, now We don't check if it's a new frame, and then We will process it again by re-updating the texture.
Not necessarily in a future PR, it's quite simple. Let me see if I have some time in the next very few days.

Reply:
Oh, ok. I assumed that it would return NULL if we already got that frame.

Reply:
Updated it. Was quite easy, now We schedule an update 4 times faster than the framerate, but the texture is updated only if a new frame is detected. As I can see, now it's smoother and consistent between runs. It wasn't scheduled for this PR, but thank you for the advice.